File: reference/spring-boot-java-gradle-hexagonal/src/main/java/com/example/demo/adapter/inbound/web/CreateWidgetRequest.java
package com.example.demo.adapter.inbound.web;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.PositiveOrZero;
import jakarta.validation.constraints.Size;

public record CreateWidgetRequest(
        @NotBlank @Size(max = 120) @Schema(example = "flux capacitor") String name,
        @PositiveOrZero @Schema(example = "3") int quantity) {}

File: reference/spring-boot-kotlin-gradle-hexagonal/src/main/kotlin/com/example/demo/adapter/inbound/web/CreateWidgetRequest.kt
package com.example.demo.adapter.inbound.web

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.PositiveOrZero
import jakarta.validation.constraints.Size

data class CreateWidgetRequest(
    @field:NotBlank
    @field:Size(max = 120)
    @field:Schema(example = "flux capacitor")
    val name: String,
    @field:PositiveOrZero
    @field:Schema(example = "3")
    val quantity: Int,
)
